boards/BMS/Src/App/App_PreChargeRelay.c

```diff
@@ -32,3 +33,32 @@ void App_PrechargeRelay_Open(const struct PrechargeRelay *const precharge_relay)
 {
     precharge_relay->open_relay();
 }
+
+bool App_PrechargeRelay_CheckFaults(
+    struct BmsCanTxInterface *can_tx,
+    bool                      is_charger_connected,
+    bool                      is_ts_rising_slowly,
+    bool                      is_ts_rising_quickly)
+{
+    static uint8_t precharge_fault_count    = 0;
+    bool           precharge_shutdown_fault = false;
+    bool           has_precharge_fault =
+        (is_charger_connected) ? is_ts_rising_slowly : (is_ts_rising_slowly | is_ts_rising_quickly);
+
+    if (has_precharge_fault)
+    {
+        precharge_fault_count++;
+    }
+    else
+    {
+        precharge_fault_count = 0;
+    }
+
+    if (precharge_fault_count == 3)
+    {
+        precharge_shutdown_fault = true;
+    }
+
+    App_CanTx_SetPeriodicSignal_PRECHARGE_FAULT(can_tx, precharge_shutdown_fault);
```

Comment:
Are we supposed to be able to recover from this fault after entering the fault state or will a power cycle be required?

Reply:
I think the plan is to require a power cycle, with the mentality that if pre-charge fails 3 times in a row, there is likely an issue that should be investigated before trying again